tests/end-to-end/external-solution-integration/pkg/testkit/connector.go

```diff
@@ -77,7 +77,7 @@ func (cc *ConnectorClient) isTokenRequestReady() error {
 		return e
 	}
 
-	if &tokenRequest.Status == nil || &tokenRequest.Status.URL == nil || tokenRequest.Status.URL == "" {
+	if tokenRequest.Status.URL == "" {
```

Comment:
Why `tokenRequest` cannot be nil?

Reply:
correct me if im wrong: the removed code does not check if tokenRequest is nil, but if the TokenRequestStatus stored in .Status is nil.
as this never was a pointer, there is just no need to check if it points to a valid address ...

Reply:
True, that the old did not check `tokenRequest` is nil. But is it possible? If possible we should check it.

Reply:
its not possible